bap/update_books.rb

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env ruby
+
+require 'json'
+
+# Files
+out_file = 'bap/book_data/AUTO_books.txt'
+ubl = 'bap/book_data/USER_ubl.txt'
+`mkdir -p bap/book_data; touch #{out_file}; touch #{ubl}`
+
+# Refresh ABL (bash)
+abl_hash = JSON.parse(
+  `curl -Ss https://raw.githubusercontent.com/openstax/content-manager-approved-books/main/approved-book-list.json`
+)
+
+# Write ABL
+File.delete(out_file)
+abl_hash['approved_books'].each do |book|
+  repo = book['repository_name']
+  book['versions'][0]['commit_metadata']['books'].each do |volume| 
+    File.write(out_file, "#{repo} #{volume['slug']}\n", mode: 'a')
```

Comment:
You could do this with jq instead of ruby if you're interested:
```bash
abl="$(curl -Ss 'https://raw.githubusercontent.com/openstax/content-manager-approved-books/main/approved-book-list.json')"
echo "$abl" | jq -r '.approved_books|.[]|.repository_name as $repo|.versions[0].commit_metadata.books|.[]|$repo + " " + .slug'
```
The output is something like

> osbooks-us-history us-history
> osbooks-microbiology microbiology
> osbooks-university-physics-bundle university-physics-volume-1
> osbooks-university-physics-bundle university-physics-volume-2
> osbooks-university-physics-bundle university-physics-volume-3
> osbooks-prealgebra-bundle prealgebra
> osbooks-prealgebra-bundle elementary-algebra
> osbooks-prealgebra-bundle intermediate-algebra

Reply:
Just formatting the 2nd line a bit:

```sh
abl="$(curl -Ss 'https://raw.githubusercontent.com/openstax/content-manager-approved-books/main/approved-book-list.json')"

query=$(cat << 'EOF'
    .approved_books                         # Select approved_books
    | .[]                                   # Foreach (approved_book) in approved_books
    | .repository_name as $repo             #   Assign $repo to be the value of .repository_name
    | .versions[0].commit_metadata.books    #   Select books in version[0].commit_metadata
    | .[]                                   #   Foreach (book) in books
    | $repo + " " + .slug                   #     Print the $repo and the .slug
EOF
)

echo "$abl" | jq -r "$query"
```